src/software/ai/hl/stp/play/penalty_kick_play.cpp

```diff
@@ -14,44 +14,61 @@ std::string PenaltyKickPlay::getName() const
 {
-    if (world.gameState().isOurPenalty())
-    {
-        return true;
-    }
-    else
-    {
-        return false;
-    }
+    return world.gameState().isOurPenalty() && !world.gameState().isStopped() &&
+           !world.gameState().isHalted();
 }
 
 void PenaltyKickPlay::getNextTactics(TacticCoroutine::push_type &yield)
 {
     auto penalty_shot_tactic = std::make_shared<PenaltyKickTactic>(
         world.ball(), world.field(), world.enemyTeam().goalie(), true);
 
+    penalty_shot_tactic->addWhitelistedAvoidArea(AvoidArea::BALL);
+    penalty_shot_tactic->addWhitelistedAvoidArea(AvoidArea::HALF_METER_AROUND_BALL);
+    penalty_shot_tactic->addWhitelistedAvoidArea(AvoidArea::ENEMY_DEFENSE_AREA);
+    penalty_shot_tactic->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+
+    auto shooter_setup_move = std::make_shared<MoveTactic>(true);
+
+    shooter_setup_move->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+    shooter_setup_move->addWhitelistedAvoidArea(AvoidArea::ENEMY_DEFENSE_AREA);
+    shooter_setup_move->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
+    shooter_setup_move->addWhitelistedAvoidArea(AvoidArea::HALF_METER_AROUND_BALL);
+
     auto move_tactic_2 = std::make_shared<MoveTactic>(true);
+    move_tactic_2->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+    move_tactic_2->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
     auto move_tactic_3 = std::make_shared<MoveTactic>(true);
+    move_tactic_3->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+    move_tactic_3->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
     auto move_tactic_4 = std::make_shared<MoveTactic>(true);
+    move_tactic_4->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+    move_tactic_4->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
     auto move_tactic_5 = std::make_shared<MoveTactic>(true);
+    move_tactic_5->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+    move_tactic_5->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
     auto move_tactic_6 = std::make_shared<MoveTactic>(true);
+    move_tactic_6->addWhitelistedAvoidArea(AvoidArea::ENEMY_HALF);
+    move_tactic_6->addWhitelistedAvoidArea(AvoidArea::FRIENDLY_HALF);
 
     do
     {
-        // Assign one robot to be the penalty shooter
-        penalty_shot_tactic->updateWorldParams(world.ball(), world.enemyTeam().goalie(),
-                                               world.field());
+        std::vector<std::shared_ptr<Tactic>> tactics_to_run;
+
+        Vector behind_ball_direction =
+            (world.ball().position() - world.field().enemyGoalpostPos()).norm();
+        Angle shoot_angle =
+            (world.field().enemyGoalpostPos() - world.ball().position()).orientation();
+
+        Point behind_ball = world.ball().position() +
+                            behind_ball_direction.norm(DIST_TO_FRONT_OF_ROBOT_METERS +
+                                                       BALL_MAX_RADIUS_METERS + 0.1);
```

Comment:
Maybe make the fudge-offset a constant here.

We should consider a 'behindBall' function since we use it so much

Reply:
noted in #968